docs/12.EXT-REFERENCE-MODULE.md

```diff
@@ -0,0 +1,177 @@
+#  Module API
+
+A JerryScript module is defined by calling the macro `JERRYX_MODULE` and supplying it with the name of the module as
+its first parameter and the name of an initialization function as its second parameter.
+
+**Note**: The `JERRYX_MODULE` macro depends on compiler support because it uses `__attribute__` extensions provided
+only by GCC and LLVM/clang to ensure that JerryScript's own module resolver (`jerryx_module_native_resolver`) will
+find a module that was declared with `JERRYX_MODULE`. If such compiler support is absent, currently
+`jerryx_module_native_resolver` will fail to find a module declared with `JERRYX_MODULE`.
```

Comment:
I would alao add a define for this feature when it is enabled, so people can test it at compile time and do a workaround if needed:

    #ifdef __GNUC__
    ...
    #define JERRYX_NATIVE_MODULES_SUPPORTED
    ...
    #endif